specification.md

```diff
@@ -795,6 +795,44 @@ Path? getStderrPath()
 
 Set/get the paths to the standard stream files.
 
+<a name="jobspec-setprelaunch"></a>
+<a name="jobspec-setpreexec"></a>
+<a name="jobspec-setpostexec"></a>
+<a name="jobspec-setpostlaunch"></a>
+```java
+void setPreLaunch(Path script)
+Path? getPreLaunch()
+void setPreExec(Path script)
+Path? getPreExec()
+void setPostExec(Path script)
+Path? getPostExec()
+void setPostLaunch(Path script)
+Path? getPostLaunch()
+```
+
+Set/get the paths to the pre/post exec/launch scripts.
+
+The `PreLaunch` script is sourced on rank 0 of the job before the parallel
+launch of the `executable`.  This script is intended for setting up the
```

Comment:
I have a terminology clash here, likely because we use the terms a bit differently in RCT?  

With `launcher`, we usually mean something like `mpirun`, `aprun`, etc,.  The job thus has no determined rank *before* launch - ranks are only a concept applicable *after* launch, and only then the qualifier (`on rank 0`) would make sense.

I appreciate the difference between running a script on `rank 0` or on all ranks.  We struggled with naming here, too, but now call `pre_exec` what runs before all ranks, and `pre_rank` what runs before a specific rank activates (we don't separate out rank 0).  

To be clear: I don't want to change anything in the spec, but want to clarify that the `pre_launch` is actually running *after* the mpirun, or whatever, well, _launches_ the job?